### PR TITLE
Use double quotes errors in 'dont mix' and 'has no attributes'

### PR DESCRIPTION
**Pull request by dixith**
### Description
This pull request fixes issue reported in #7445  for `dont mix` and `has no attributes` error messages

## Test Plan
Updated test cases for the message. New test cases not added.

### CODE REVIEW

**Comment by dixith** [comment]
@JelleZijlstra 
```
# E: eq must be "True" if order is "True"
```

Please let me know if you feel double quotes are not needed for the above error message

File: mypy/plugins/attrs.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def _determine_eq_order(ctx: 'mypy.plugin.ClassDefContext') -> bool:
     order = _get_decorator_optional_bool_argument(ctx, 'order')
 
     if cmp is not None and any((eq is not None, order is not None)):
-        ctx.api.fail("Don't mix `cmp` with `eq' and `order`", ctx.reason)
+        ctx.api.fail('Don\'t mix "%s" with "%s" and "%s"' % (cmp, eq, order), ctx.reason)
```
Comment (hauntsaninja) [inline mypy/plugins/attrs.py:215]: This change makes the error message worse. We shouldn't be interpolating the values for these arguments here.
```suggestion
        ctx.api.fail('Don\'t mix "cmp" with "eq" and "order"', ctx.reason)
```

File: mypy/plugins/attrs.py [hunk 2]
```diff
@@ -226,7 +226,7 @@ def _determine_eq_order(ctx: 'mypy.plugin.ClassDefContext') -> bool:
         order = eq
 
     if eq is False and order is True:
-        ctx.api.fail("eq must be True if order is True", ctx.reason)
+        ctx.api.fail('eq must be "True" if order is "True"', ctx.reason)
```
Comment (hauntsaninja) [inline mypy/plugins/attrs.py:229]: No need for quotes here, I think just plain True reads more clearly as a Python literal

File: mypy/plugins/attrs.py (hunk 1 above)
Comment (dixith) [reply mypy/plugins/attrs.py:215]: @hauntsaninja Below code looks good? 
```python
ctx.api.fail('Don\'t mix "{}" with "{}" and "{}"'.format(cmp, eq, order), ctx.reason)
```

Comment (dixith) [reply mypy/plugins/attrs.py:215]: or should i change it to 
```python
ctx.api.fail('Don\'t mix {} with {} and {}'.format(cmp, eq, order), ctx.reason)
```

File: mypy/plugins/attrs.py [hunk 3]
```diff
@@ -226,7 +226,7 @@ def _determine_eq_order(ctx: 'mypy.plugin.ClassDefContext') -> bool:
         order = eq
 
     if eq is False and order is True:
-        ctx.api.fail("eq must be True if order is True", ctx.reason)
+        ctx.api.fail('eq must be True if order is True', ctx.reason)
```
Comment (JelleZijlstra) [inline mypy/plugins/attrs.py:229]: You could undo this quote change but I don't feel strongly

**Review by JelleZijlstra — APPROVED** [review]

**Comment by dixith** [comment]
@hauntsaninja Can you please review and approve the PR? JelleZijlstra has approved.

**Comment by JelleZijlstra** [comment]
(Sorry, I forgot to click merge. I don't think Shantanu needs to review too; these changes are pretty simple.

**Pull request merged**